### DISCUSSION TITLE
Change curve type from quadratic to cubic

### DISCUSSION THREAD

**Discussion by RandomMaerks** [jacobxperez/lexica-ultralegible#12] [Ideas]
At the moment, when a glyph has curves, they're quadratic Bézier curves.

![image](https://github.com/jacobxperez/atkinson-hyperlegible-pro/assets/109415614/276aef52-46e8-4893-8119-d26297f79dd3)

It might be due to the fact that .ttf files of the [original Atkinson Hyperlegible fonts](https://github.com/googlefonts/atkinson-hyperlegible) were used to convert to .sfd files. TrueType fonts use quadratic curves.

Quadratic curves are usually fine, but they are a pain to edit. Unlike cubic Bézier curves where one curve has 2 end points and the control points are separated, quadratic curves have multiple "intermediate points" and 1 control point for every two consecutive points in order to compensate for that smoothness.
You can edit a cubic curve just by editing either the end points or the control points, but you'll have to edit a quadratic curve by either manually editing every single end points (which doesn't always turn out great), or by merging all the points and be left with a "set smoothness".

---

So my idea is to a) _**change the curve type to cubic**_ and b) _**manually edit each curve to be cubic**_. It sounds like a terrible idea, but it might save a lot of headaches in the future.

![image](https://github.com/jacobxperez/atkinson-hyperlegible-pro/assets/109415614/297225cd-1db1-4247-b175-a4ada5dc1678)

You can go to Element → Font Info → Layers, then in the Font Type section, click 'All layers cubic'.
![image](https://github.com/jacobxperez/atkinson-hyperlegible-pro/assets/109415614/34282654-d083-4d41-af92-faf2515abddc)

Once you've done that, you can merge the 'intermediate points' and fix the control points by harmonising the curves.

https://github.com/jacobxperez/atkinson-hyperlegible-pro/assets/109415614/6d2c08fc-c38a-4585-a419-5e23a008a839

**Comment by RandomMaerks** [comment]
@jacobxperez If you approve of this idea, I can try to implement this alongside the Vietnamese language support thing

**Reply by jacobxperez** [reply]
@RandomMaerks Yes you are right I will start working on this implementation immediately, thanks for the awesome contribution you are bringing to this project.

**Reply by RandomMaerks** [reply]
Great! Also I will be implementing this for the Regular style since I'm also adding Vietnamese simultaneously. So if you don't mind, you can start with the other styles

**Reply by jacobxperez** [reply]
@RandomMaerks Alright thanks!!
